### PR TITLE
Cab book

### PR DESCRIPTION
**Pull request by Shilpa-73**
1) camelCase fields name fixed for all models
2) daos reviewed and created for bookings and cabs
=> This will returns a past bookings and nearest cabs
3) customer can book the cab 

Pending
- Authentication

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,14 +2,18 @@ import express from 'express';
 import dotenv from 'dotenv';
```
Comment (alichherawalla) [inline index.js:2]: add `dist` to .gitignore

File: server/daos/customers.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+export const checkCabAvailability = async (cabId)=>{
+    //Check already reserved this requested cab
+    let cabReserved = await db.bookings.findOne({
+        attributes:['vehicleId','id','amount'],
+        where:{
+            status : 'CAB_ASSIGNED',
+            vehicleId:cabId,
+            endTime:null,
+            createdAt: {
+                [Op.gt]: new Date().setHours(0, 0, 0, 0),
```
Comment (alichherawalla) [inline server/daos/customers.js]: please use moment consistently everywhere

File: server/daos/customers.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                    {
+                        model: db.passport,
+                        as:'vehicle_category',
+                    }
+                ]
+            })
+        );
+
+        if(withPassword){
+            await db.passport.findOne({
```
Comment (alichherawalla) [inline server/daos/customers.js]: what does this line do?

File: server/daos/customers.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    convertDbResponseToRawResponse(
+        await db.customers.findOne({
+            where
+        })
+)
+
+export const bookCabs = async({cabId,...rest})=>{
+    let response = {}
+    try{
+        response.bookignData = await db.bookings.create({
```
Comment (alichherawalla) [inline server/daos/customers.js]: response.bookignData = await db.bookings.create(rest)

File: server/database/dbUtils.js [hunk 5]
```diff
@@ -34,3 +39,76 @@ export const deletedId = new GraphQLObjectType({
   name: 'Id',
   fields: () => ({ id: { type: GraphQLNonNull(GraphQLInt) } })
 });
+
+export const findOneById = async (model, args) => {
```
Comment (alichherawalla) [inline server/database/dbUtils.js:43]: if its findOneById we can just assume id will be the second parameter

File: server/gql/customMutations/authLogin.js [hunk 6]
```diff
@@ -51,52 +41,49 @@ export const loginMutation = {
...
-            let passportData = await passport.findOne({
-                where: {
-                    user_id: existCustomer.id,
-                    user_type: 'CUSTOMER',
-                    provider_type: 'LOCAL'
-                },
-                raw: true
+            let passportData = await findOneByCriteria(db.passport, {
+                userId: existCustomer.id,
+                userType: 'CUSTOMER',
```
Comment (alichherawalla) [inline server/gql/customMutations/authLogin.js]: use constants for CUSTOMER and LOCAL

File: server/daos/customers.js (hunk 3 above)
Comment (Shilpa-73) [reply server/daos/customers.js]: This is just one boolean flag for getting the password from another table, If passed true then we only return password data

**Pull request merged**